.golangci.yml

```diff
@@ -0,0 +1,252 @@
+    # '\t' is counted as 1 character by default, and can be changed with the tab-width option
+    line-length: 110
+    # tab width in spaces. Default to 1.
+    tab-width: 2
+  maligned:
+    # print struct with more effective memory layout or not, false by default
+    suggest-new: true
+  misspell:
+    # Correct spellings using locale preferences for US or UK.
+    # Default is to use a neutral variety of English.
+    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
+    locale: UK
+    ignore-words:
+      - degitx
+  nakedret:
+    # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
+    max-func-lines: 1
+  nolintlint:
+    # Enable to ensure that nolint directives are all used. Default is true.
+    allow-unused: false
+    # Disable to ensure that nolint directives don't have a leading space. Default is true.
+    allow-leading-space: false
+    # Exclude following linters from requiring an explanation.  Default is [].
+    allow-no-explanation: []
+    # Enable to require an explanation of nonzero length after each nolint directive. Default is false.
+    require-explanation: true
+    # Enable to require nolint directives to mention the specific linter being suppressed. Default is false.
+    require-specific: true
+  unparam:
+    # Inspect exported functions, default is false. Set to true if no external program/library imports your code.
+    # XXX: if you enable this setting, unparam will report a lot of false-positives in text editors:
+    # if it's called for subdir of a project it can't find external interfaces. All text editor integrations
+    # with golangci-lint call it on a directory with the changed file.
+    check-exported: false
+  whitespace:
+    multi-if: false   # Enforces newlines (or comments) after every multi-line if statement
+    multi-func: false # Enforces newlines (or comments) after every multi-line function signature
+  gofumpt:
+    # Choose whether or not to use the extra rules that are disabled
+    # by default
+    extra-rules: false
+  errcheck:
+    # report about not checking of errors in type assertions: `a := b.(MyStruct)`;
+    # default is false: such cases aren't reported by default.
+    check-type-assertions: true
+
+    # report about assignment of errors to blank identifier: `num, _ := strconv.Atoi(numStr)`;
+    # default is false: such cases aren't reported by default.
+    check-blank: true
+  unused:
+    # treat code as a program (not a library) and report unused exported identifiers; default is false.
+    # XXX: if you enable this setting, unused will report a lot of false-positives in text editors:
+    # if it's called for subdir of a project it can't find funcs usages. All text editor integrations
+    # with golangci-lint call it on a directory with the changed file.
+    check-exported: false
+  testpackage:
+    # regexp pattern to skip files
+    skip-regexp: degitx_test\.go
```

Comment:
@stepanvalyavskiy why do we skip test file?

Reply:
Because it is in top level package degitx, and this linter requires all test to be in other package. But we can create degitx_test under degitx package. OK.

Reply:
@stepanvalyavskiy I see now, I thought it's a suppression for all linters in test file. I think it makes sense to ignore all test files for this rule. Something like `.+_test\.go`

Reply:
`testpackage` is a linter that is needed only to care that all tests are in separate *_test packages. we should then disable this linter. Please check [it](https://github.com/maratori/testpackage#motivation) and conclude whether we should disable `testpackage`  linter at all.

Reply:
@stepanvalyavskiy yes, I think we can disable this liter globally